tools/engine_tool/lib/src/commands/lint_command.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io' show File;
+
+import 'package:path/path.dart' as p;
+
+import '../dart_utils.dart';
+import '../proc_utils.dart';
+import 'command.dart';
+import 'flags.dart';
+
+/// The root 'lint' command.
+final class LintCommand extends CommandBase {
+  /// Constructs the 'lint' command.
+  LintCommand({
+    required super.environment,
+  }) {
+    _linterArgs['dart'] = <String>[findDartBinDirectory(environment)];
+    _linterRunner['java'] = p.join(findDartBinDirectory(environment), 'dart');
+  }
+
+  // Map from lint-type to path of tool to invoke.
+  // Paths are relative to 'engine/src/flutter'.
+  static const Map<String, String> _linters = <String, String>{
+    'dart': 'ci/analyze.sh',
+    'python': 'ci/pylint.sh',
+    'c': 'ci/clang_tidy.sh',
+    'java': 'tools/android_lint/bin/main.dart',
+  };
```

Comment:
These won't work on Windows. Consider making `et lint` exit early if ran on a Windows machine.

Reply:
Are there Windows equivalents we could add?

Reply:
Hm doesn't look like it. We'll need to add corresponding batch scripts or convert them to Dart.

Reply:
kk, for now I will disable the lints on Windows and leave a TODO for you.

Reply:
FYI, clang tidy runs on Windows as a git hook using Git Bash which emulates bash on Windows... that might be another option to unblock Windows.